decision-records/008-reset-the-javascript-api.md

```diff
@@ -0,0 +1,72 @@
+# Reset the JavaScript API for our components
+
+## Decision
+
+We will mark all class and instance properties other than the constructor and the init method as deprecated as part of a v4.x release.
+
+In v5.0 we will then add JSDoc annotations to mark the deprecated properties as private.
+
+In future releases we will then gradually introduce a new public JavaScript API for components, which is:
+
+- based on the needs of users
+- clearly defined and documented
+- comprehensively tested
+
+**Date:** 2023-07-20
+
+**Status:** Accepted
```

Comment:
**issue** These need lifting to the top of the document (after the title) for consistency with the other decision records.